src/sema/functions.rs

```diff
@@ -552,8 +552,11 @@ pub fn contract_function(
     fdecl.selector = selector;
 
     if func.ty == pt::FunctionTy::Constructor {
-        // In the eth solidity, only one constructor is allowed
-        if ns.target == Target::EVM {
+        // In the eth solidity and Substrate, only one constructor is allowed
+        //
+        // The Substrate runtime supports multiple constructors, but they also need to have a name.
+        // Currently, we only support either naming the constructors or overloading them, but not both.
```

Comment:
I am confused. Overloading constructors requires multiple constructors. Are we allowing multiple constructors?

Reply:
You are right it is a bit confusing. What I meant to say here is that the solang compiler itself does not support the combination of both.